File: src/main/java/github/macro/pieces/Archer.java
package github.macro.pieces;

import github.macro.Direction;
import github.macro.players.Player;
import org.jetbrains.annotations.NotNull;

import java.util.List;

/**
 * Created by Macro303 on 2021-Jun-02.
 */
public class Archer extends Piece {
	public Archer(@NotNull Player player) {
		super(player, 6, "A", List.of(
				Direction.NORTH,
				Direction.EAST,
				Direction.SOUTH,
				Direction.WEST
		));
	}

	@Override
	public String toString() {
		return "Archer{} " + super.toString();
	}
}

File: src/main/kotlin/github/macro/pieces/Archer.kt
package github.macro.pieces

import github.macro.Direction
import github.macro.players.Player

/**
 * Created by Macro303 on 2018-02-12.
 */
class Archer(player: Player) : Piece(
	player = player,
	distance = 6,
	symbol = "A",
	directions = arrayOf(Direction.NORTH, Direction.EAST, Direction.SOUTH, Direction.WEST)
) {
//	private static final String symbol = "\uD83C\uDFF9";
//	private static final String symbol = "🏹";
//	private static final String symbol = "A";

	override fun toString(): String = "Archer() ${super.toString()}"
}